common/services/user-locations.js

```diff
@@ -0,0 +1,64 @@
+const axios = require('axios')
+
+const { AUTH_PROVIDERS, NOMIS_ELITE2_API } = require('../../config')
+const { decodeAccessToken } = require('../lib/access-token')
+const referenceDataService = require('./reference-data')
+
+function getUserLocations(token) {
+  return _getAgencies(token).then(agencies => {
+    return Promise.all(
+      agencies.map(agencyId =>
+        referenceDataService.getLocations({ nomisAgencyId: agencyId })
+      )
+    ).then(locations => locations.map(results => results[0].id))
+  })
+}
+
+function _getAgencies(token) {
+  let agenciesPromise
+
+  const tokenData = decodeAccessToken(token)
+
+  switch (tokenData.auth_source) {
+    case 'auth':
+      agenciesPromise = _getAuthAgencies(token)
```

Comment:
Could just return within the switch to avoid needed to declare the promise:

```js
function _getAgencies(token) {
  const { auth_source: source } = decodeAccessToken(token)

  switch (source) {
    case 'auth':
      return _getAuthAgencies(token)
      break
    case 'nomis':
      return _getNomisAgencies(token)
      break
  }
}
```

Also could destructure the token.

Reply:
Also just thinking about fallback, maybe we should have a `default` case which returns a successful promise incase the source doesn't exist and it doesn't cause an unhandled promise error.